### PR TITLE
Adjust practice worlds to make use of new wavefield model envelope

### PR DESCRIPTION
**Pull request by M1chaelM**
This PR adjusts the wavefield parameters for the practice worlds to cover a broader spectrum of possibilities using our recently updated wavefield envelope (see #684). The changes introduced to the worlds are as follows:
* I updated all easy (0) worlds so they have no waves at all.
* I changed all intermediate (1) worlds to use period 6, gain 0.6, which produces mild swells. I also varied the direction of the wavefield so teams will not expect it to always stay the same.
* I changed the wavefield of some of the hard (2) worlds so they are not all the same. Specifically, I left the period of 2 and gain of 2.0 for tasks that rely on perception. For other tasks, such as stationkeeping and wayfinding I used period 6, gain 0.8 to produce medium swells. I also introduced some simple variations to the direction of the wavefield.

## To test:
* The easy worlds can probably be verified with a spot check and visual inspection of the code.
* Try to run a few of the intermediate worlds to make sure the wave effects are reasonable and nothing is too challenging. Please also look to make sure I set all the parameters to the values given above.
* For the hard worlds, please run each and verify that they are challenging but not so extreme as to be unsolvable, and not broken (i.e. no weird physical effects).

### CODE REVIEW

**Comment by tejalbarnwal** [comment]
Hey, i observed some of the things for difficult worlds(_2 worlds):
1. For the *station-keeping world*, in the absence of any vehicle control, the vehicle, after some time, collides with the beach. 
I have attached a [video](https://drive.google.com/file/d/1Gb3woYHs-3AwkVT32OHRvG66v-Mp9drq/view?usp=sharing) simulation(please watch it at 2x speed). I think this won't cause an issue when teams start applying controls but I am concerned about the extent of control that would be required to tackle this.
A sort of similar situation arises with *wayfinding task2* as well, where without control, the robot rests against the beach. [video](https://drive.google.com/file/d/1rOUQNPiT2XscMi2MV5kdsT_Q7v0V326Q/view?usp=sharing)
2. for the *wildlife2 task*, the crocodile and turtle do get submerged inside the water periodically. Since teams rely on pose topics to know the position of wildlife, this shouldn't be an issue ig, right?
3. With *acoustic_perception2*, *acoustic_tracking2* and *scan_dock_deliver2*, the vehicle drifts along the direction of the waves and wind, which I believe is a natural behavior(coz there is no application of controls). Eg: [video](https://drive.google.com/file/d/1hPXnWFssPfrKlbrqFzu0mEDnpkqXSEjG/view?usp=sharing)
4. Finally, in case of *perception_task2*, I observed buoys to bounce up when they are spawned in the environment. I wanted to confirm, if this is expected behaviour?[video](https://drive.google.com/file/d/1I9rqZZtar2HTLbsxE-P1k91p39vUwznS/view?usp=sharing)

**Comment by M1chaelM** [comment]
> * For the _station-keeping world_, in the absence of any vehicle control, the vehicle, after some time, collides with the beach.
>   I have attached a [video](https://drive.google.com/file/d/1Gb3woYHs-3AwkVT32OHRvG66v-Mp9drq/view?usp=sharing) simulation(please watch it at 2x speed). I think this won't cause an issue when teams start applying controls but I am concerned about the extent of control that would be required to tackle this.
>   A sort of similar situation arises with _wayfinding task2_ as well, where without control, the robot rests against the beach. [video](https://drive.google.com/file/d/1rOUQNPiT2XscMi2MV5kdsT_Q7v0V326Q/view?usp=sharing)
> * for the _wildlife2 task_, the crocodile and turtle do get submerged inside the water periodically. Since teams rely on pose topics to know the position of wildlife, this shouldn't be an issue ig, right?
> * With _acoustic_perception2_, _acoustic_tracking2_ and _scan_dock_deliver2_, the vehicle drifts along the direction of the waves and wind, which I believe is a natural behavior(coz there is no application of controls). Eg: [video](https://drive.google.com/file/d/1hPXnWFssPfrKlbrqFzu0mEDnpkqXSEjG/view?usp=sharing)
> * Finally, in case of _perception_task2_, I observed buoys to bounce up when they are spawned in the environment. I wanted to confirm, if this is expected behaviour?[video](https://drive.google.com/file/d/1I9rqZZtar2HTLbsxE-P1k91p39vUwznS/view?usp=sharing)

Thanks for this very helpful and detailed review. These observations all sound like they are within the range of expected behavior, but I will double check especially the stationkeeping and wayfinding worlds. I agree with your assessment of the easy and medium worlds. I think if you could just visually inspect the code to make sure the parameters are consistent, and maybe try out 1-2 easy and 1-2 medium worlds chosen at random that would be enough.

I'm not sure what you mean by "after 5-6 hours." Do you mean the next time you'll be available to review is in 5-6 hours? If so, that's fine, and thanks again for taking a look.

**Comment by tejalbarnwal** [comment]
>I'm not sure what you mean by "after 5-6 hours." Do you mean the next time you'll be available to review is in 5-6 hours? If so, that's fine, and thanks again for taking a look.

Hey, the `follow_path0` world had a non-zero gain value. I updated its value to 0.0

File: vrx_gz/worlds/2023_practice/practice_2023_follow_path0_task.sdf [hunk 1]
```diff
@@ -502,14 +502,14 @@
           key: "gain"
           value {
             type: DOUBLE
-            double_value: 0.3
```
Comment (tejalbarnwal) [inline vrx_gz/worlds/2023_practice/practice_2023_follow_path0_task.sdf:505]: i updated this to 0

**Comment by M1chaelM** [comment]
> 1. For the _station-keeping world_, in the absence of any vehicle control, the vehicle, after some time, collides with the beach.
>    I have attached a [video](https://drive.google.com/file/d/1Gb3woYHs-3AwkVT32OHRvG66v-Mp9drq/view?usp=sharing) simulation(please watch it at 2x speed). I think this won't cause an issue when teams start applying controls but I am concerned about the extent of control that would be required to tackle this.
>    A sort of similar situation arises with _wayfinding task2_ as well, where without control, the robot rests against the beach. [video](https://drive.google.com/file/d/1rOUQNPiT2XscMi2MV5kdsT_Q7v0V326Q/view?usp=sharing)

I watched your videos and also replayed the hard stationkeeping and wayfinding tasks. The behavior you're seeing is consistent with what we've done in previous competitions, so I think it's OK. In fact, I asked the same question the first time I saw the same thing, and the consensus was that it is actually fairly realistic, and not too uncommon in real-world competitions. If a team leaves their platform in the water without control it can float away and sometimes that means it ends up back on the beach.

**Comment by M1chaelM** [comment]
> 2\. for the _wildlife2 task_, the crocodile and turtle do get submerged inside the water periodically. Since teams rely on pose topics to know the position of wildlife, this shouldn't be an issue ig, right?

I looked at this more closely and I will pose the question to the group. On the one hand, this effect also seems physically realistic to me (e.g. sometimes crocodiles go under water), but it would also be fine to reduce the gain a little bit. I made a gif to show the effect at the current setting of period 2, gain 2:

wildlife_period2_gain2

Here's what it would look like at period 6, gain 0.8:
wildlife_period6_gain0_8

I think this second one looks a little more realistic and is sufficiently challenging, so I will switch to that for now.

**Comment by M1chaelM** [comment]
> * With _acoustic_perception2_, _acoustic_tracking2_ and _scan_dock_deliver2_, the vehicle drifts along the direction of the waves and wind, which I believe is a natural behavior(coz there is no application of controls). Eg: [video](https://drive.google.com/file/d/1hPXnWFssPfrKlbrqFzu0mEDnpkqXSEjG/view?usp=sharing)

Yes, this looks fine.

> * Hey, the _follow_path0_ world had a non-zero gain value. I updated its value to 0.0.

Thanks!

> * I noticed a similar bouncing effect in _perceptation_task1_ as I did in _perception_task2_. The bouncing of balls and buoys was also present here, but I found it to be more pronounced in this case. [video](https://drive.google.com/file/d/1MF3FR2wqwITyBUoxWmgJtWs-ydPN-i2c/view?usp=sharing)

Yeah, good observation; this does not look good. It is expected that the buoys should drop into the water and then float back up, but they should not bounce off the surface like basketballs, so something is a bit weird here. I don't think this is a problem with this task or the environmental settings, so it shouldn't block this PR. However, it would be worth presenting your second video to the technical team and creating an issue to look into possible problems with our buoyancy plugin.

**Comment by M1chaelM** [comment]
Let me know if I've missed anything. If the above comments address your concerns, could you please approve this PR?

**Review by tejalbarnwal — APPROVED** [review]

**Comment by tejalbarnwal** [comment]
> Let me know if I've missed anything. If the above comments address your concerns, could you please approve this PR?

I guess we have covered it all. Sure.

**Pull request merged**